### PR TITLE
Modularized models

### PR DESCRIPTION
**Pull request by dandorman**
Right now, Tanker doesn't handle modularized models. This is a quick fix to make that happen. I ripped the `constantize` code straight from Rails' ActiveSupport::Inflector.

### CODE REVIEW

**Comment by dandorman** [comment]
I think really the only pertinent revision is the last one, actually: 8bfec33

**Comment by kidpollo** [comment]
This is great work but can we put the modularized classes and the did you mean in a single pull request so I can merge it automagically and not conflict whith the facets stuff which is creat but I want to hold on a little bit on it for a while because I am also looking into it!

**Comment by dandorman** [comment]
There's actually a problem with the did-you-mean feature, as I discovered when I tried using it against actual data. The problem is that IndexTank tries to help you out too much.

Let's say I'm searching a BlogPost model for _bunnny_ (note the third _n_ in _bunny_). Tanker generates a query something like this: `__any:(bunnny) __type:(BlogPost)`. When IndexTank returns its results, the `didyoumean` looks like `__any:(bunny) __type:(BlogPost)`. This is great if what you want to do is fire off a subsequent query to IndexTank with the corrected query, but in the given state, it's useless for displaying to the end user. So my rspec examples are wrong.

This isn't insurmountable; it wouldn't be very difficult to parse IndexTank's results for user display. It just seemed like more trouble than it was worth at the time, so I haven't done any more work on it yet.

The other issue with the did-you-mean feature is that IndexTank only uses the special `text` field for calculating it. (Actually, I believe that's the case for a lot of the extra features--I'm pretty sure snippets work the same way.)

Right now, Tanker uses the `__any` field as its catchall field. I propose we use `text` instead, to leverage IndexTank's cooler features. It is of course possible for users to define `text` themselves in their `tankit` blocks; but then that gets added to `__any` too, and it becomes all too easy to exceed IndexTank's 100k-per-document limit. I've found that out the hard way :)

Consequently, if you look at 93d1aa1338587b5cfb92689df364e7fad8dc2bae, I added the ability to configure a different "catchall" field.  The default is still `__any`, so existing indexes won't be affected, but users who want it can set it to `text`.

So for the time being, I reverted both those changes (e1e347c988d20f7ca7e0ed2e14131d53beb10b08), because I didn't feel like it was ready. I'm happy to do a little more work if you think it's worth it though.

**tl;dr** There are some problems with my did-you-mean stuff right now.

**Comment by kidpollo** [comment]
k thanks for the info! ok let me have a conversation with the index tank guys about this did you mean feature may be we can wait and get a better idea on how this work and how it will continue to work in the future. In the mean time can I take the modularized classes commit only? Does need anything in particular from other commit. Can you cherry pick only that commit and rebase onto master and submit the pull request by itself?

**Comment by dandorman** [comment]
I'm gonna close this, 'cause it's been superseded by the other, more streamlined pull request.
